File: src/main/java/architecture/abnormalities/common/entity/ordeals/violet/IOrdealsVioletEntity.java
package architecture.abnormalities.common.entity.ordeals.violet;

import architecture.abnormalities.common.entity.ordeals.IOrdealsEntity;
import architecture.abnormalities.init.tag.AbnormalitiesEntityTags;
import architecture.goldenboughs_lib.api.world.entity.ai.CampHurtByTargetGoal;
import net.minecraft.world.entity.Entity;

public interface IOrdealsVioletEntity extends IOrdealsEntity {
	@Override
	default boolean isCamp(Entity entity) {
		return IOrdealsEntity.super.isCamp(entity) || entity.getType().is(AbnormalitiesEntityTags.ORDEALS_VIOLET);
	}

	@Override
	default void registerGoals() {
		IOrdealsEntity.super.registerGoals();
		getTargetSelector().addGoal(2, new CampHurtByTargetGoal(getMob(), IOrdealsVioletEntity.class));
	}
}

File: src/main/kotlin/architecture/abnormalities/common/entity/ordeals/violet/IOrdealsVioletEntity.kt
package architecture.abnormalities.common.entity.ordeals.violet

import architecture.abnormalities.common.entity.ordeals.IOrdealsEntity
import architecture.abnormalities.init.tag.AbnormalitiesEntityTags
import architecture.goldenboughs_lib.api.world.entity.ai.CampHurtByTargetGoal
import net.minecraft.world.entity.Entity

interface IOrdealsVioletEntity : IOrdealsEntity {
	override fun isCamp(entity: Entity): Boolean {
		return super.isCamp(entity) || entity.type.`is`(AbnormalitiesEntityTags.ORDEALS_VIOLET)
	}

	override fun registerGoals() {
		super.registerGoals()
		targetSelector.addGoal(2, CampHurtByTargetGoal(mob, IOrdealsVioletEntity::class.java))
	}
}
